src/app_service/service/community/service.nim

```diff
@@ -519,6 +519,13 @@ QtObject:
     # checking length is sufficient - communityTokensMetadata list can only extend
     return communityTokens.len != prevCommunityTokens.len
 
+  proc updateMemberRole(self: Service, community, prevCommunity: CommunityDto) =
+    if community.memberRole != prevCommunity.memberRole:
```

Comment:
Unfortunately, you can't compare memberRole like this. This would work only if YOU lost or received ownership. Because this parameter tells what is your member role in the community. But if you want to catch that some other member lost or received owner role then you need to compare who is the owner in _community_ and in _prev_.
If the owner is different then you need to update both: old owner and new owner.

Reply:
> Unfortunately, you can't compare memberRole like this. This would work only if YOU lost or received ownership. Because this parameter tells what is your member role in the community. But if you want to catch that some other member lost or received owner role then you need to compare who is the owner in _community_ and in _prev_.
> 
> If the owner is different then you need to update both: old owner and new owner.

Hi there, thanks for your comment @endulab 

> This would work only if YOU lost or received ownership.

This is my intent here. Because the same code is used on emit and reception, this code will be triggered only if the current client lost or gained ownership. I am handling the other cases separately.

Reply:
Why separately? Isn't it the same algorithm?

Reply:
You are right, this should be handled at the same place. Let me have a look at this.